source/include/FreeRTOS_DNS_Callback.h

```diff
@@ -27,37 +27,46 @@
 
 
 #ifndef FREERTOS_DNS_CALLBACK_H
-#define FREERTOS_DNS_CALLBACK_H
+    #define FREERTOS_DNS_CALLBACK_H
 
-#if ( ( ipconfigDNS_USE_CALLBACKS == 1 ) && ( ipconfigUSE_DNS != 0 ) )
+    #ifdef __cplusplus
+        extern "C" {
+    #endif
 
 /* FreeRTOS includes. */
     #include "FreeRTOS.h"
 
 /* FreeRTOS+TCP includes. */
     #include "FreeRTOS_IP.h"
+    #include "FreeRTOSIPConfigDefaults.h"
 
     #include "FreeRTOS_DNS_Globals.h"
 
 /* Standard includes. */
     #include <stdint.h>
```

Comment:
I think we should move this to the top. Before FreeRTOS.h is included.

Reply:
The file includes some default value, only needed if they are not defined in FreeRTOS.h